Document environment variables (#143)
benbusby/whoogle-search: README.md

  - Ubuntu: `sudo apt-get install -y libcurl4-openssl-dev libssl-dev`
  - Arch: `pacman -S curl openssl`

> ## Environment variables

Reviewer: ```suggestion
## Environment Variables
```

> ## Environment variables
> You may use environment variables to customize your Whoogle instance:
> 
> | Variable           | Description                                                    |
> | ------------------ | -------------------------------------------------------------- |
> | WHOOGLE_USER       | The username for basic auth. WHOOGLE_PASS should also be used. |
> | WHOOGLE_PASS       | The password for basic auth. WHOOGLE_USER should also be used. |
> | WHOOGLE_PROXY_USER | The username of the proxy server.                              |
> | WHOOGLE_PROXY_PASS | The password of the proxy server.                              |
> | WHOOGLE_PROXY_TYPE | The type of the proxy server. For example "socks5".           |
> | WHOOGLE_PROXY_LOC  | The location of the proxy server (host or ip).                 |
> | EXPOSE_PORT        | The port where Whoogle will be exposed.                        |
> | HTTPS_ONLY         | Enforce HTTPS. Se the setion below.                            |
> 

Reviewer: A couple of spelling fixes and rewording here. Basic auth requires both user/pass to be set, so fixed the wording there as well to make that a bit more clear.

```suggestion
## Environment Variables
There are a few optional environment variables available for customizing a Whoogle instance:

| Variable           | Description                                                    |
| ------------------ | -------------------------------------------------------------- |
| WHOOGLE_USER       | The username for basic auth. WHOOGLE_PASS must also be set if used. |
| WHOOGLE_PASS       | The password for basic auth. WHOOGLE_USER must also be set if used. |
| WHOOGLE_PROXY_USER | The username of the proxy server.                              |
| WHOOGLE_PROXY_PASS | The password of the proxy server.                              |
| WHOOGLE_PROXY_TYPE | The type of the proxy server. Can be "socks5", "socks4", or "http".           |
| WHOOGLE_PROXY_LOC  | The location of the proxy server (host or ip).                 |
| EXPOSE_PORT        | The port where Whoogle will be exposed.                        |
| HTTPS_ONLY         | Enforce HTTPS. (See [here](https://github.com/benbusby/whoogle-search#https-enforcement))                            |

```

docker run --publish 5000:5000 --detach --name whoogle-search whoogle-search:1.0
```

> You may want to set the environment:

Reviewer: ```suggestion
Optionally, you can also enable some of the following environment variables to further customize your instance:
```

Note: You should have your own domain name and [an https certificate](https://letsencrypt.org/getting-started/) in order for this to work properly.

- Heroku: Ensure that the `Root URL` configuration on the home page begins with `https://` and not `http://`
- Docker build: Add `--build-arg use_https=1` to your run command

> - Docker image: st the environment variable HTTPS_ONLY=1

Reviewer: ```suggestion
- Docker image: Set the environment variable HTTPS_ONLY=1
```
